spec/e2e/end_to_end.rb

```diff
@@ -6,62 +6,115 @@
-      data_hash = JSON.parse(data_string)
-      data = Coinbase::Wallet::Data.from_hash(data_hash)
-      w2 = u.import_wallet(data)
-      expect(w2).not_to be_nil
-      puts "Imported wallet with ID: #{w2.id}, default address: #{w2.default_address}"
-
-      puts 'Listing wallet addresses...'
-      addresses = w2.addresses
-      expect(addresses.length).to be > 1
-      puts "Listed addresses: #{addresses.map(&:to_s).join(', ')}"
-
-      puts 'Fetching wallet balances...'
-      balances = w2.balances
-      expect(balances.length).to be >= 1
-      puts "Fetched balances: #{balances}"
-
-      puts 'Transfering 1 Gwei from default address to second address...'
-      a1 = addresses[0]
-      a2 = addresses[1]
-      t = a1.transfer(1, :gwei, a2).wait!
-      expect(t.status).to eq('complete')
-      puts "Transferred 1 Gwei from #{a1} to #{a2}"
-
-      puts 'Fetching updated balances...'
-      first_balance = a1.balances
-      second_balance = a2.balances
-      expect(first_balance[:eth]).to be > BigDecimal('0')
-      expect(second_balance[:eth]).to be > BigDecimal('0')
-      puts "First address balances: #{first_balance}"
-      puts "Second address balances: #{second_balance}"
+      e2e_test
     end
   end
+
+  describe 'use serve signer' do
+    it 'behaves as expected' do
+      # Use Server-Signer only half the runs to save test time.
+      skip if rand >= 0.5
+      e2e_test(use_server_signer: true)
+    end
+  end
+end
+
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/AbcSize
+def e2e_test(use_server_signer: false)
+  # GitHub secrets truncate newlines as whitespace, so we need to replace them.
+  # See https://github.com/github/docs/issues/14207
+  api_key_name = ENV['API_KEY_NAME'].gsub('\n', "\n")
+  api_key_private_key = ENV['API_KEY_PRIVATE_KEY'].gsub('\n', "\n")
+
+  # Use default API URL if not provided
+  api_url = ENV['API_URL']
+
+  Coinbase.configure do |config|
+    config.api_key_name = api_key_name
+    config.api_key_private_key = api_key_private_key
+    config.api_url = api_url if api_url
+    config.use_server_signer = use_server_signer
+  end
+
+  puts 'Fetching default user...'
+  u = Coinbase.default_user
+  expect(u).not_to be_nil
+  puts "Fetched default user with ID: #{u.id}"
+
+  if use_server_signer
+    signer = Coinbase::ServerSigner.default
+    puts "Using ServerSigner with ID: #{signer.id}"
+  end
+
+  data_string = ENV['WALLET_DATA']
+  expect(data_string).not_to be_nil
+
+  puts 'Creating new wallet...'
+  w1 = u.create_wallet
+  expect(w1).not_to be_nil
+  puts "Created new wallet with ID: #{w1.id}, default address: #{w1.default_address}"
+
+  puts 'Creating new address...'
+  new_address = w1.create_address
+  expect(new_address).not_to be_nil
+  puts "Created new address with ID: #{new_address.id} in wallet with ID #{w1.id}"
+
+  # To move funds from imported wallet, do not use Server-Signer.
+  Coinbase.configuration.use_server_signer = false
```

Comment:
@jazz-cb can we split up these test cases a bit more?

Since there is no "importing" for server signers, lets instead just fetch a known server-signer backed wallet that is funded, and use that wallet for the server signer case?

Reply:
@alex-stone did some refactoring here for the tests. not having an enumerator makes it difficult to find a specific wallet for server-signer. so i am using import_wallet logic for now, but use a wallet created by signer. so the import data has no seed